saleor/saleor
# Process `handle-promotion-toggle` task in batches.

I want to merge this change because it processes `handle-promotion-toggle` task in batches, what prevents Celery queue getting stuck in case of big number of starting/ending promotions.

Test case:
1. Run saleor db, api, redis and worker in Docker
2. Create 1000 promotions, with single rule and single variant. Set start date = now().
3. Observe worker output and/or `Promotion.last_notification_scheduled_at` records.

AC:
1. 1000 promotions should be handled in 10 batches (100 promotions per each batch)
2. Every next batch should be processed in ~5s interval 
3. After processing 10 batches, check if all promotion records have `promotion.last_notification_scheduled_at != Null`




# Impact


# Docs





- [ ] Link to documentation:

# Pull Request Checklist



- [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
- [ ] Database migrations are either absent or optimized for zero downtime
- [ ] The changes are covered by test cases
- [ ] All new fields/inputs/mutations have proper labels added (`ADDED_IN_X`, `PREVIEW_FEATURE`, etc.)
- [ ] All migrations have proper dependencies
- [ ] All indexes are added concurrently in migrations
- [ ] All RunSql and RunPython migrations have revert option defined
